pkg/handlers/config.go

```diff
@@ -64,6 +66,77 @@ type CurrentAppConfigResponse struct {
 	ConfigGroups []kotsv1beta1.ConfigGroup `json:"configGroups"`
 }
 
+type DownloadAppConfigResponse struct {
+	Success bool   `json:"success"`
+	Error   string `json:"error,omitempty"`
+}
+
+func (h *Handler) DownloadAppConfig(w http.ResponseWriter, r *http.Request) {
+	downloadAppConfigResponse := DownloadAppConfigResponse{
+		Success: false,
+	}
+
+	foundApp, err := store.GetStore().GetAppFromSlug(mux.Vars(r)["appSlug"])
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to get app from app slug"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	sequence, err := strconv.Atoi(mux.Vars(r)["sequence"])
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to parse app sequence"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	filename := mux.Vars(r)["filename"]
+	if filename == "" {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to parse filename, parameter was empty"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	archiveDir, err := ioutil.TempDir("", "kotsadmconfig")
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to create temp directory"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+	defer os.RemoveAll(archiveDir)
+
+	configValue, err := getAppConfigValueForFile(foundApp, int64(sequence), filename, archiveDir)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to get app config"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	decoded := make([]byte, 0)
+	_, err = base64.StdEncoding.Decode(decoded, configValue)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to decode config value"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+
+	reader := bytes.NewReader(decoded)
+	w.Header().Set("Content-Type", "application/octet-stream")
+	_, err = io.Copy(w, reader)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to copy file to client"
+		JSON(w, http.StatusBadRequest, downloadAppConfigResponse)
+		return
+	}
+
+	JSON(w, http.StatusOK, DownloadAppConfigResponse{Success: true})
```

Comment:
this should not be called here, and io.Copy should not be used before headers are written.  this function should call these directly
```
	w.WriteHeader(http.StatusOK)
	w.Write(decoded)
```